### PR TITLE
rewrite table components to use hooks, add pagination

### PR DESCRIPTION
**Pull request by jharting**
I know, I know, this is huge. It is however impossible to switch to hooks without rewriting an entire component :-/

### CODE REVIEW

**Comment by jharting** [comment]
@karelhala I am thinking that if the hook definitions here catch on we may eventually move them to frontend components for them to be reusable. Please review especially `src/hooks/table.js`

**Comment by jharting** [comment]
related https://github.com/patternfly/patternfly-react/pull/1503

File: src/components/RemediationTable.js [hunk 1]
```diff
@@ -36,148 +35,164 @@ function formatDate (date) {
...
+                            }]
+                        }
+                        rows={ rows }
+                        { ...sorter.props }
+                        { ...selector.props }
+                    >
+                        <TableHeader/>
+                        <TableBody/>
+                    </Table> :
+                    <p className='ins-c-remediations-table--empty'>No Playbooks found</p>
```
[the remark this answers is not in the document]
Comment (jharting) [reply src/components/RemediationTable.js]: @ryelo good catch! updated

File: src/hooks/table.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import intersection from 'lodash/intersection';
+
+import { SEARCH_DEBOUNCE_DELAY } from '../constants';
+
+const chain = (f1, f2) => (...args) => {
+    f1(...args);
+    f2(...args);
+};
+
+function callbacks () {
```
Comment (karelhala) [inline src/hooks/table.js]: Phew his function is really hard o understand can you add some comments what does it do?

Comment (jharting) [reply src/hooks/table.js]: @karelhala thanks for feedback. I added a comment

**Pull request merged**